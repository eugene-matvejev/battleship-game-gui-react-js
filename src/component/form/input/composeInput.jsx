import React from 'react';
import PropTypes from 'prop-types';

export default (Component) => {
    const ComposedInput = ({ className, label, errors, ...props }) =>
        <div className={`form-input ${className}`}>
            <label className={`form-input_label`}>
                {label}
                <Component {...props} />
                {
                    errors &&
                    errors.length &&
                    <ul className="form-input_errors">
                        {errors.map((v, i) => <li className="form-input_error" key={i}>{v}</li>)}
                    </ul>
                }
            </label>
        </div>;

    ComposedInput.propTypes = {
        className: PropTypes.string,
        label: PropTypes.string,
        errors: PropTypes.arrayOf(PropTypes.string),
    };
    ComposedInput.defaultProps = {
        className: '',
    };

    return ComposedInput;
}
