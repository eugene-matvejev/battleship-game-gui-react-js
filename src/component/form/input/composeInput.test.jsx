import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import composeInput from './composeInput';

configure({ adapter: new Adapter() });

const ComposedInput = composeInput((props) => <input {...props} />);

describe('<ComposedInput/>', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<ComposedInput {...props} />);

            expect(c).toMatchSnapshot();
        });
    });
});
