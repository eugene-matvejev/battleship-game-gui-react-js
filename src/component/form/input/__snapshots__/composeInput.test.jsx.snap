// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposedInput/> render with default/required props 1`] = `
<div
  className="form-input "
>
  <label
    className="form-input_label"
  >
    <Component />
  </label>
</div>
`;
